openwisp_controller/config/base/config.py

```diff
@@ -459,6 +463,9 @@ def _check_changes(self):
             'backend', 'config', 'context', 'status'
         ).get(pk=self.pk)
         for attr in ['backend', 'config', 'context']:
+            if attr == 'backend' and getattr(self, attr) != getattr(current, attr):
+                # storing old backend to send backend change signal after save
+                self._old_backend = current.backend
```

Comment:
Why is this piece of code added into this loop and not just outside of it? Eg:

```python
if self.backend != current.backend:
    self._old_backend = current.backend
```

I don't see the benefit in having this code in this loop.